Result of days

koder95 opened the pull request.

koder95 committed:
Last task files copy

koder95 committed:
Add unit tests for core service and model classes

Introduces comprehensive JUnit tests for Main, IO (serializers/deserializers), model implementations, and transaction service/handlers. These tests cover various scenarios including null handling, file operations, and business logic validation to ensure correctness and robustness of the application's core functionality.

slade13 commented on src/test/java/core/basesyntax/MainTest.java:
This test exercises Main. The task explicitly says don’t test the Main class. Remove the class entirely.

slade13 commented on src/test/java/core/basesyntax/io/DeserializersTest.java:
```diff
@@ -0,0 +1,70 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+public class DeserializersTest {
```
Real filesystem coupling - you create and delete real files (manual `Files.deleteIfExists`). Prefer `@TempDir` to avoid path clashes and to make tests OS-agnostic.

slade13 commented on src/test/java/core/basesyntax/io/SerializersTest.java:
```diff
@@ -0,0 +1,95 @@
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        boolean readOnly = file.setWritable(false);
```
Using `File#setWritable(false)` is unreliable across OS/filesystems. Use @TempDir and try to write to a directory (should fail) or to a path under a removed directory to force an exception.

slade13 commented on src/test/java/core/basesyntax/service/impl/TransactionHandlersTest.java:
```diff
@@ -0,0 +1,166 @@
+    @BeforeAll
+    static void setUp() {
+        MapTransactionServiceTest.setUp();
+        SERVICE = MapTransactionServiceTest.instance();
```
`TransactionHandlersTest `appears to rely on the `MapTransactionServiceTest `singleton (INSTANCE) pattern. Tests should not depend on other test classes. Build your handler or service map locally in each class’ `@BeforeEach`.

slade13 reviewed (changes requested):
You want to merge your branch `main `into remotes branch `main`:

> [koder95](https://github.com/koder95)
> wants to merge 2 commits into
> [mate-academy:main](https://github.com/mate-academy/jv-fruit-shop-tests/tree/main)
> from
> [koder95:main](https://github.com/koder95/jv-fruit-shop-tests/tree/main)

Please create a separate feature branch in your fork and then create a PR from your feature branch. Please also check my comments and fix these problems accordingly.

koder95 committed:
Delete MainTest.java

koder95 committed:
Tests not depends on themselfs

Removed the static instance() method from MapTransactionServiceTest and updated TransactionHandlersTest to initialize MapTransactionService directly with handlers. This improves test isolation and clarity by avoiding shared state between tests.

koder95 committed:
Refactor tests to use JUnit TempDir for file operations

Updated DeserializersTest and SerializersTest to use JUnit's @TempDir for managing temporary files and directories, improving test isolation and reliability. Removed manual file cleanup and replaced static file paths with dynamically generated temporary paths. Added tests for handling directory inputs.

koder95 commented:
> You want to merge your branch `main `into remotes branch `main`:
> 
> > [koder95](https://github.com/koder95)
> > wants to merge 2 commits into
> > [mate-academy:main](https://github.com/mate-academy/jv-fruit-shop-tests/tree/main)
> > from
> > [koder95:main](https://github.com/koder95/jv-fruit-shop-tests/tree/main)
> 
> Please create a separate feature branch in your fork and then create a PR from your feature branch. Please also check my comments and fix these problems accordingly.

@slade13 How can I create branch and bind it with mate academy platform? I created [new branch](https://github.com/koder95/jv-fruit-shop-tests/tree/my-solution) and [pull request](https://github.com/mate-academy/jv-fruit-shop-tests/pull/1208) but I had to reopen this PR, because system did not notice the change.

slade13 reviewed (changes requested):
Please create PR from feature branch, not from `main`.

koder95 commented:
> Please create PR from feature branch, not from main.

I already did that yesterday:

> @slade13 How can I create branch and bind it with mate academy platform? <ins>**I created [new branch](https://github.com/koder95/jv-fruit-shop-tests/tree/my-solution) and [pull request](https://github.com/mate-academy/jv-fruit-shop-tests/pull/1208)**</ins> but I had to reopen this PR, because system did not notice the change.

How can I bind [that PR](https://github.com/mate-academy/jv-fruit-shop-tests/pull/1208) to mate.academy? Should I close this PR? What's next? I created new PR. Should I close that too?

Please tell me what I should do?
